### PR TITLE
CircleCI/Jenkins: use buildkit

### PR DESCRIPTION
**Pull request by thaJeztah**
~follow-up to https://github.com/docker/cli/pull/1599~ rebased

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,2 +1,6 @@
+.dockerignore
```
Comment (silvin-lubecki) [inline .dockerignore]: That's super meta 😸

Comment (thaJeztah) [reply .dockerignore]: 😂 yes

Comment (thaJeztah) [reply .dockerignore]: I usually add the Dockerfile itself as well, didn't take time to look into which Dockerfile may be used docker-in-docker

File: Jenkinsfile [hunk 2]
```diff
@@ -5,8 +5,9 @@ wrappedNode(label: 'linux && x86_64', cleanWorkspace: true) {
 
     stage "Run end-to-end test suite"
     sh "docker version"
+    sh "docker info"
     sh "E2E_UNIQUE_ID=clie2e${BUILD_NUMBER} \
         IMAGE_TAG=clie2e${BUILD_NUMBER} \
-        make -f docker.Makefile test-e2e"
+        make DOCKER_BUILDKIT=1 -f docker.Makefile test-e2e"
```
Comment (tiborvass) [inline Jenkinsfile]: never again makefile variables. Use `DOCKER_BUILDKIT=1 make ...` instead.

Comment (tiborvass) [reply Jenkinsfile]: I fixed it

**Comment by tiborvass** [comment]
TestSigProxyWithTTY timed out twice in a row.

**Comment by thaJeztah** [comment]
> TestSigProxyWithTTY timed out twice in a row.

Yes. It's flaky for some reason, but I'm unable to reproduce it locally 😞 almost seems machine-specific (help welcome!)

**Comment by thaJeztah** [comment]
`TestSigProxyWithTTY` was fixed in https://github.com/docker/cli/pull/2016. Rebased this one, and all is green now; merging

**Pull request merged**